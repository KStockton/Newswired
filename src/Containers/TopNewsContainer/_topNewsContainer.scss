.TopNews-header {
  box-shadow: 2px 2px 2px #cab6b6;
  font-family: 'Noto Sans', sans-serif;
  font-size: 1.7rem;
  padding: 36px 0px;
}

.TopNews-wrapper {
  display: flex;
  overflow: scroll;
  width: 100vw;
  background-image: linear-gradient( rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.5) ),url('../../Assets/coffee-cup.jpg'); 
  background-repeat: no-repeat;
  background-position-x: center;
  background-position-y: center;
  background-size: cover;
  overflow-y: hidden;
}