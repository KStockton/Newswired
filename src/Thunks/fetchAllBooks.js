import * as action from '../actions/index'



export const fetchAllBooks = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      const books = await response.json()
      dispatch(action.isLoading(false));
      dispatch(action.getAllBooks(books))
    } catch (error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}