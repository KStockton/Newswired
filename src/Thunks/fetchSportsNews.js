import * as action from '../actions/index';


export const fetchSportsNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      dispatch(action.getSportsNews(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'sportsNews'))
    }
  }
}