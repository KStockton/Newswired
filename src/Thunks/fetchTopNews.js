import * as action from '../actions/index';


export const fetchTopNews = (url) => {

  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      dispatch(action.getTopNews(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message, 'topNews'))
    }
  }
}

