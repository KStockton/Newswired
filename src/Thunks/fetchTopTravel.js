import * as action from '../actions/index';


export const fetchTopTravel = (url) => {
console.log('url', url)
  return async (dispatch) => {
    try {
      dispatch(action.isLoading(true))
      const response = await fetch(url)
      if(!response.ok) {
      throw Error(response.sendText)
    }
      const data = await response.json()
      console.log('data', data)
      dispatch(action.getTopTravel(data))
      dispatch(action.isLoading(false))
    } catch(error) {
      dispatch(action.hasErrored(error.message))
    }
  }
}

