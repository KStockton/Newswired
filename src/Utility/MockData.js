export const uncleanBooks = {
  results: {
    books: [{
              title: 'Becoming', 
              description: 'The former first lady describes how she balanced work',
              primary_isbn10: 1234,
              weeks_on_list: 20,
              book_image: 'cool.com',
              rank: 1,
              contributor: 'by Michelle Obama',
              dagger: 0
            }]
  }
}

export const cleanBooks = [{
  id: 1234,
  rank: 1,
  author: 'by Michelle Obama',
  bookImage: 'cool.com',
  description: 'The former first lady describes how she balanced work',
  title: 'Becoming',
  weeksOnList: 20,
  isFavorited: false,
}]

export const mockTopNews = [
  {
    title: 'Syria',
    id: 232
  }, {
    titel: 'Trump loses hair',
    id: 45
  }]

  export const mockNewsError = ''

  export const mockRealNewsError = 'something went wrong'

  export const mockTravel = [{title:'New York'},{title:'Boston'}]

  export const mockRealTravelError = 'No Travel Here'