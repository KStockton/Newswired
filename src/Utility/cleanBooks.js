export const cleanBooks = (uncleanBooks) => {
 const cleanedBooks = uncleanBooks.results.books.map(book => {
    return {
        rank: book.rank,
        author: book.author,
        bookImage: book.book_image,
        description: book.description,
        title: book.title
      }
  })
  return cleanedBooks
}