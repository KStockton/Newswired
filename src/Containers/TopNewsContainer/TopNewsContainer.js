import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchTopNews } from '../../Thunks/fetchTopNews';
import PropTypes from 'prop-types';
import Loading from '../../Components/Loading/Loading';
import {Link} from 'react-router-dom';
import Card from '../../Components/Card/Card';
const shortid = require('shortid');


export class TopNewsContainer extends Component {

  componentDidMount() {
    if (this.props.allTopNews.length === 0){
      const topNewsUrl = `https://newsapi.org/v2/top-headlines?country=us&apiKey=${process.env.REACT_APP_NEWSAPI_API_KEY}`;
      this.props.fetchTopNews(topNewsUrl);
    }
  }

  displayTopNews =() => {
    return this.props.allTopNews.map(location => {
      return (<Card {...location} key={shortid.generate()}/>);
    });
  };
  
  render() {
    const {error, allTopNews}= this.props;
    const errorMsg = 
                    <div>
                      <h2>{error}</h2>
                      <Link to="/Options">Back</Link>
                    </div>;
    const displayTopNews = this.displayTopNews();
    
    return (
      <section>
        <h2 className="TopNews-header">Top News</h2>
        <div className="TopNews-wrapper Card-wrapper">
          {(error !== '' && allTopNews.length === 0 ) && errorMsg}
          {(error === '' && allTopNews.length === 0 ) && <Loading/>}
          {displayTopNews}
        </div>
      </section>
    );
  }
}

TopNewsContainer.propTypes = {
  allTopNews: PropTypes.array,
  error: PropTypes.string,
  fetchTopNews: PropTypes.func
};

export const mapStateToProps = (state) => ({
  allTopNews: state.allTopNews,
  error: state.error
});

export const mapDispatchToProps = (dispatch) => ({
  fetchTopNews: (url) => dispatch(fetchTopNews(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(TopNewsContainer);
