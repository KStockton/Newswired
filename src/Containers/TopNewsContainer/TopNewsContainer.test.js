import React from 'react';
import { shallow } from 'enzyme';
import * as mockData from './../../Utility/MockData';
import { fetchTopNews } from '../../Thunks/fetchTopNews'
import { TopNewsContainer, mapStateToProps, mapDispatchToProps } from './TopNewsContainer';
jest.mock('../../Thunks/fetchTopNews')

describe('TopNewsContainer', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<TopNewsContainer
      allTopNews={mockData.mockTopNews}
      error = {mockData.mockNewsError}
      fetchTopNews={fetchTopNews}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should render loading if data is in the process of fetching', () => {
    const allTopNews = []
    const error = mockData.mockNewsError
    wrapper.setProps({allTopNews})
    wrapper.setProps({error})
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  });

  it('should render an error if an error message is passed', () => {
    const allTopNews = []
    const error = mockData.mockRealNewsError
    wrapper.setProps({allTopNews})
    wrapper.setProps({error})
    wrapper.update()
    expect(wrapper).toMatchSnapshot()
  });

  it('should call displayTopNews when rendered', () => {
    wrapper.instance().displayTopNews = jest.fn()
    wrapper.instance().render()
    expect(wrapper.instance().displayTopNews).toHaveBeenCalled()
  });

  it('should render 2 topNews articles when 2 books are passed', () => {
   let result = wrapper.instance().displayTopNews().length
    expect(result).toBe(2)
  })

});