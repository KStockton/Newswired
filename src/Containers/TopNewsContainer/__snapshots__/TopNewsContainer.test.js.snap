// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNewsContainer should match the snapshot 1`] = `
<section>
  <h2
    className="TopNews-header"
  >
    Top News
  </h2>
  <div
    className="TravelContainer-wrapper Card-wrapper"
  >
    <Card
      id={232}
      key="yoYXvwbP9"
      title="Syria"
    />
    <Card
      id={45}
      key="kf1ZefxwkG"
      titel="Trump loses hair"
    />
  </div>
</section>
`;

exports[`TopNewsContainer should render an error if an error message is passed 1`] = `
<section>
  <h2
    className="TopNews-header"
  >
    Top News
  </h2>
  <div
    className="TravelContainer-wrapper Card-wrapper"
  >
    <div>
      <h2>
        something went wrong
      </h2>
      <Link
        to="/Options"
      >
        Back
      </Link>
    </div>
  </div>
</section>
`;

exports[`TopNewsContainer should render loading if data is in the process of fetching 1`] = `
<section>
  <h2
    className="TopNews-header"
  >
    Top News
  </h2>
  <div
    className="TravelContainer-wrapper Card-wrapper"
  >
    <Loading />
  </div>
</section>
`;
